validate/validate.py

```diff
@@ -88,21 +71,18 @@ def run_validate(args):
 
     all_files_pass = True
 
-    validation_function = validate_file
-    if args.type in DIR_TYPES:
-        validation_function = validate_dir
-
     for path in [Path(pathname) for pathname in args.path]:
         try:
-            validation_function(path, args.type)
+            file_type, file_extension = detect_file_type_and_extension(path)
```

Comment:
Eventually we should rewrite `detect_file_type_and_extension` and `check_compressed` to use https://pypi.org/project/python-magic/

Reply:
Definitely! I created an issue for this